pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1260,7 +1260,7 @@ async function resolveDependency (
         err.pkgsStack = getPkgsInfoFromIds(options.parentIds, ctx.resolvedPkgsById)
         return err
       },
-      updateToLatest: options.updateToLatest,
+      updateToLatest: options.update && options.updateToLatest,
```

Comment:
Looks a bit confusing. Maybe should be done in some other place? To have the correct updateToLatest at this point.

Reply:
I agree, it's not very elegant, and I did not have many good ideas. At face value, `updateToLatest` functions like a flag most of the time, but `requestPackage` treats it as its own update path.

I made a change now in 5f7ee0328f90537b213c71542e71eade22924aec, to make the `update` option of `resolveValue` more explicit, so that `updateToLatest` will be correct. That at least keeps any future use of that function from falling into this trap. What do you think?